pkg/controller/storagecluster/reconcile.go

```diff
@@ -409,3 +456,63 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 
 	return resources
 }
+
+// newStorageClassDeviceSets converts a list of StorageDeviceSets into a list of Rook StorageClassDeviceSets
+func newStorageClassDeviceSets(storageDeviceSets []ocsv1.StorageDeviceSet, topologyMap ocsv1.NodeTopologyMap) []rook.StorageClassDeviceSet {
+	var storageClassDeviceSets []rook.StorageClassDeviceSet
+
+	for _, ds := range storageDeviceSets {
+		resources := ds.Resources
+		if resources.Requests == nil && resources.Limits == nil {
+			resources = defaults.DaemonResources["osd"]
+		}
+
+		topologyKey := ds.TopologyKey
+		topologyKeyValues := []string{}
+		noPlacement := ds.Placement.NodeAffinity == nil && ds.Placement.PodAffinity == nil && ds.Placement.PodAntiAffinity == nil
+
+		if topologyKey != "" && noPlacement {
+			topologyKey, topologyKeyValues = topologyMap.GetKeyValues(topologyKey)
+		}
+
+		for i := 0; i <= ds.Replica; i++ {
```

Comment:
just double checking this is right. or should it be ```i < ds.Replica``` ?

Reply:
....it is not right! Good catch.